metaseq/checkpoint_utils.py

```diff
@@ -327,31 +326,77 @@ def _is_checkpoint_sharded(checkpoint_files) -> bool:
     return sd["cfg"]["distributed_training"]["use_sharded_state"]
 
 
-def get_paths_to_load(local_path, suffix="rank-"):
-    checkpoint_files = glob(re.sub(f"{suffix}[0-9]+", f"{suffix}*", local_path))
-    if not _is_checkpoint_sharded(checkpoint_files):
+def _cache_checkpoint_files(path: str, suffix):
+    """
+    Given a checkpoint path, first try to expand it according to
+    a specified filename pattern. If `path` doesn't match the pattern
+    we search the remote for it as-is. Then cache all matching files
+    and return the local paths.
+
+    Ex. 1:
+        Remote: [path://checkpoint_last-shard0.pt, path://checkpoint_last-shard1.pt]
+        Input: path=remote://path/checkpoint_last-shard0.pt, suffix=shard
+        Output: [/local/checkpoint_last-shard0.pt, /local/checkpoint_last-shard1.pt]
+
+    Ex. 2:
+        Remote: [path://checkpoint_last-model_part-0.pt, path://checkpoint_last-model_part-1.pt]
+        Input: path=remote://path/checkpoint_last-model_part-0.pt, suffix=shard
+        Output: [/local/checkpoint_last-model_part-0.pt]
+
+    Ex. 3:
+        Remote: [path://checkpoint_last.pt]
+        Input: path=remote://path/checkpoint_last-shard0.pt, suffix=shard
+        Output: []
+    """
+    src_dir = os.path.dirname(path)
+    path_prefix = re.sub(rf"{suffix}[0-9]+\.pt$", f"{suffix}*", path)
```

Comment:
Could you add a comment here for what this checks? Looks like this is handling the scenario when we don't have shards, just model part files?

Reply:
Maybe an example similar to the examples in the docstring